### ISSUE TITLE
Error 500 on almost every page

### ISSUE THREAD

**Issue by AlexKalopsia** [linuxserver/docker-codimd#13]
I just started using CodiMD and I have one issue, where all of the Intro page links lead to 500 error. It's the same if I open my profile and click on Features. I am wondering if this is just a normal result of self-hosting, in which case it would still be nice to make sure those link keep pointing at the CodiMD website rather than the selfhosted address.

Broken Links:
- Real time collaboration
- Works with charts and MathJax
- Supports slide mode
- Releases (footer)
- Profile -> Features

## Expected Behavior
These links should not be changed with the local ip but lead to the official CodiMD website

## Current Behavior
The links point to myip/xxxxx resulting in not finding any of those pages

## Environment
Running with docker-compose on Synology NAS

## Command used to create docker container (run/create/compose/screenshot)
```
  codimd:
    container_name: codimd
    image: linuxserver/codimd:latest
    volumes:
      - /volume1/docker/codimd/config:/config
    environment:
      - DB_HOST=myip
      - DB_USER=codimd
      - DB_PASS=psw
      - DB_NAME=db
      - DB_PORT=3306
      - PGID=1000
      - PUID=1000
      - TZ=Europe/Stockholm
    ports:
      - 3001:3000
    depends_on:
      - mariadb
    restart: unless-stopped
```

**Comment by thelamer** [comment]
Slide mode works for me along with profile=> features, though I can replicate a 500 with the release notes. 

There is an md for it in /config/docs so it should be loading. 

This link functions on the stock `nabo.codimd.dev/hackmdio/hackmd:1.4.1` so we need to figure out what is different and why. 
@CHBMB

**Comment by CHBMB** [comment]
@AlexKalopsia 

As mentioned here, unless you're running behind a reverse proxy, you have to run this on port 3000.

https://github.com/linuxserver/docker-codimd#application-setup

That being said, just looking at this now I have a setup working perfectly using tag `1.4.0-ls11` but when I upgrade to `latest` I can reproduce the errors and my working setup shows up as offline.

We need to take a look at things.

**Comment by AlexKalopsia** [comment]
Just to clarify I am, in fact, behind reverse proxy 3001:3000

**Comment by thelamer** [comment]
@CHBMB and I have been able to narrow this down to a datatype thing with Mysql and some problems with postgres also: 
```
2020-01-14T00:22:04.535Z error:     Incorrect string value: '\xE1\xBB\x93ng ...' for column `database`.`Notes`.`content` at row 1
2020-01-14T00:22:04.535Z error:         Incorrect string value: '\xE1\xBB\x93ng ...' for column `database`.`Notes`.`content` at row 1
2020-01-14T00:22:04.547Z info:     172.17.0.1 - - [14/Jan/2020:00:22:04 +0000] "GET /s/release-notes HTTP/1.1" 500 - 
```
Specifically the collation/column encoding in your DB needs to be set to `utf8mb4` I believe

**Comment by thelamer** [comment]
honestly your best bet for right now would be to use the `1.4.0-ls11` tag for now as @CHBMB has noted unless you want to modify your database setup.

**Comment by HerHde** [comment]
I can reproduce this; Both over the LinuxServer.io Let'sEncrypt proxy over an internal Docker network as well as connecting to the port directly. Using a `docker-compose.yml` based on the example, running on a Ubuntu 18.04 VM using `Docker version 18.09.7, build 2d0083d`  and

```
docker-compose version 1.17.1, build unknown
docker-py version: 2.5.1
CPython version: 2.7.17
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
```


```
error:  Incorrect string value: '\xE2\x9C\x94|\xE2\x9C...' for column `codimd`.`Notes`.`content` at row 1
error:  Incorrect string value: '\xE2\x9C\x94|\xE2\x9C...' for column `codimd`.`Notes`.`content` at row 1
info:   172.23.0.2 - - [17/Feb/2020:14:24:43 +0000] "GET /features HTTP/1.1" 500 - "-" "Mozilla/5.0 [...]
```

Changing the tag to `1.4.0-ls11`, rebuilding and even deleting and rebuilding the DB yielded a similar result:
```
error:  ER_TRUNCATED_WRONG_VALUE_FOR_FIELD: Incorrect string value: '\xE2\x9C\x94|\xE2\x9C...' for column `codimd`.`Notes`.`content` at row 1
error:  ER_TRUNCATED_WRONG_VALUE_FOR_FIELD: Incorrect string value: '\xE2\x9C\x94|\xE2\x9C...' for column `codimd`.`Notes`.`content` at row 1
info:   172.23.0.2 - - [17/Feb/2020:14:31:39 +0000] "GET /features HTTP/1.1" 500 - "-" "Mozilla/5.0 
```

I tried the latest version again with a new document, adding the `/feature`-content paragraph by paragraph until it failed with the [first table](https://demo.codimd.org/#Permissions). It seems like the unicode chars are causing the problem, I cloud easily reproduce the failure by inserting any like ✔, ★ or 🤔 into an empty document:

![](https://user-images.githubusercontent.com/7630662/74665370-de553f00-519f-11ea-9fb5-80e036436aa0.png)

And again:

```
Incorrect string value: '\xF0\x9F\xA4\x94' for column `codimd`.`Notes`.`content` at row 1
```

Commenting out the `DB_HOST` to use SQLite works fine, so, @thelamer was right about the encoding. He already provided a viable solution to set the encoding here https://github.com/linuxserver/docker-codimd/issues/6#issuecomment-511548799 which he also already sent upstream: https://github.com/codimd/server/issues/134. 👏 Thank you, good work!

## TL;DR:

For me it helped to append

```
collation-server = utf8_unicode_ci
init-connect='SET NAMES utf8'
character-set-server = utf8
```

to the `[mysqld]` and 

```
default-character-set=utf8
```

to the `[mysql]` block in my mariadb `custom.cnf` before initialising the DB as stated in https://github.com/linuxserver/docker-codimd/issues/6#issuecomment-511548799.
